v3/src/components/slider/slider-thumb.tsx

```diff
@@ -25,8 +25,10 @@ export const CodapSliderThumb = observer(function CodapSliderThumb({sliderContai
   const downOffset = useRef(0)
 
   useEffect(() => {
+    if (sliderModel.value <= sliderModel.axis.min) sliderModel.setValue(sliderModel.axis.min)
+    if (sliderModel.value >= sliderModel.axis.max) sliderModel.setValue(sliderModel.axis.max)
```

Comment:
This doesn't seem like the right place to do this. Validating/updating the model value should be handled by the model not by the React component at render time. What are you trying to fix/achieve here?